core/domain/story_services.py

```diff
@@ -342,11 +342,26 @@ def _save_story(committer_id, story, commit_message, change_list):
             'Unexpected error: received an invalid change list when trying to '
             'save story %s: %s' % (story.id, change_list))
 
+    topic = topic_fetchers.get_topic_by_id(
```

Comment:
Do we need to fetch the topic again on line 382? (Also, should we move the check on line 384 over here?)

Reply:
Done, removed extra fetch and moved around checks.